vllm/executor/openvino_executor.py

```diff
@@ -0,0 +1,539 @@
+    """
+
+    def __init__(
+        self,
+        model_config: ModelConfig,
+        parallel_config: ParallelConfig,
+        scheduler_config: SchedulerConfig,
+        device_config: DeviceConfig,
+        lora_config: Optional[LoRAConfig] = None,
+        kv_cache_dtype: Optional[str] = "auto",
+    ) -> None:
+        self.model_config = model_config
+        self.parallel_config = parallel_config
+        self.scheduler_config = scheduler_config
+        self.device_config = device_config
+        self.lora_config = lora_config
+
+        # Load FP32 models to OpenVINO
+        model_config.dtype = torch.float32
+
+        self.model_runner = ModelRunner(model_config,
+                                        parallel_config,
+                                        scheduler_config,
+                                        device_config,
+                                        lora_config=self.lora_config,
+                                        kv_cache_dtype=kv_cache_dtype,
+                                        is_driver_worker=True)
+        # Uninitialized cache engine. Will be initialized by self.init_cache_engine().
+        self.cache_config = None
+        self.cache_engine = None
+        self.gpu_cache = None
+        self.cpu_cache = None
+
+        # Not required for OpenVINO, but without torch.distributed initialization all CC operations failed
+        self._init_distributed_environment()
+
+    def init_model(self) -> None:
+        # Initialize the model.
+        set_random_seed(self.model_config.seed)
+        # TODO: ilavreno: here we also need to measure current amount of available GPU memory in case of GPU inference
+        # it will later be used to understand real
+        self.init_gpu_memory = 0
+
+    def load_model(self):
+        self.model_runner.load_model()
+
+    def profile_num_available_blocks(
+        self,
+        block_size: int,
+        device_memory_utilization: float,
+        cpu_swap_space: int,
+        cache_dtype: str,
+    ) -> Tuple[int, int]:
+        """Profiles the peak memory usage of the model and returns the maximum
+        number of Accelerator and CPU cache blocks that can be allocated.
+
+        Args:
+            block_size: The size of the cache block.
+            device_memory_utilization: The fraction of the total device memory to use.
+            cpu_swap_space: The size of the CPU swap space in bytes.
+        """
+        num_gpu_blocks, num_cpu_blocks = 0, 0
+        cache_block_size = OpenVINOCacheEngine.get_cache_block_size(
+            block_size, cache_dtype, self.model_config, self.parallel_config, self.device_config)
+
+        num_cpu_blocks = int(cpu_swap_space // cache_block_size)
+        num_cpu_blocks = max(num_cpu_blocks, 0)
+
+        # TODO: ilavreno: currently we cannot specify GPU as device
+        # so, the code below is dead, but we can change the condition and test it
+        if self.device_config.device.type == 'gpu':
```

Comment:
@vladimir-paramuzov
Here we have some profiling code to understand amount of GPU device memory which can be used for KV cache. 
This branch is not tested